### PR TITLE
Issue #1059 Fix And Some Debug Overlay Changes

### PR DESCRIPTION
**Pull request by majordwarf**
### Contains

Fixes Issue #1059 and some extra changes in debug overlay to reduce the clutter.

### How to test

Enable Debug Mode > Try enabling Wireframe or Bounding Box mode. You'll get a overlay text to specify the active mode.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

File: engine/src/main/java/org/terasology/logic/players/DebugControlSystem.java [hunk 1]
```diff
@@ -45,7 +45,7 @@
     private WorldProvider world;
     
     @In
-    private Config config;
+    private Config  config;
```
Comment (costinvisan) [inline engine/src/main/java/org/terasology/logic/players/DebugControlSystem.java]: extra spae

File: engine/src/main/java/org/terasology/rendering/nui/layers/ingame/DebugInfo.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import org.terasology.rendering.nui.WidgetUtil;
+
+public class DebugInfo extends CoreScreenLayer {
+
+    @Override
+    public void initialise() {
+        WidgetUtil.trySubscribe(this, "closeButton", widget -> getManager().closeScreen(DebugInfo.this));
+    }
+
+}
```
Comment (costinvisan) [inline engine/src/main/java/org/terasology/rendering/nui/layers/ingame/DebugInfo.java]: No new line at the end of the file

Comment (costinvisan) [inline engine/src/main/java/org/terasology/rendering/nui/layers/ingame/DebugInfo.java]: No newline at the end of the file

File: engine/src/main/java/org/terasology/logic/players/DebugControlSystem.java (hunk 1 above)
Comment (majordwarf) [reply engine/src/main/java/org/terasology/logic/players/DebugControlSystem.java]: Ah, didn't noticed this. Thanks for pointing out, I'll edit and revert with a new commit.

**Comment by majordwarf** [comment]
Sorry for two commits. I forgot to add a new line at end of one of the file.

**Comment by Cervator** [comment]
No problem and thanks! Going to trigger our build bot

@GooeyHub: add to whitelist

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by jellysnake** [comment]
For future reference, we don't stress about how many commits there are in a PR.  
Often times if the feature/change is hefty it's easier to review when it's logically split up into commits.

If you make a ton of commits with one line changes, we can also just squash and merge, which will remove the high commit count.

**Pull request merged**